internal/models/user.go

```diff
@@ -730,10 +730,20 @@ func (u *User) SoftDeleteUserIdentities(tx *storage.Connection) error {
 	return nil
 }
 
-func obfuscateFieldForSoftDelete(field string) string {
-	if field != "" {
-		softDeleteId, _ := crypto.GenerateNanoId(5)
-		return fmt.Sprintf("%s-%x", softDeleteId, sha256.Sum256([]byte(field)))
-	}
-	return field
+func obfuscateValue(id uuid.UUID, value string) string {
+	hash := sha256.Sum256([]byte(id.String() + value))
+	return base64.RawURLEncoding.EncodeToString(hash[:])
+}
+
+func obfuscateEmail(u *User, email string) string {
+	return obfuscateValue(u.ID, email)
+}
+
+func obfuscatePhone(u *User, phone string) string {
+	// Field converted from VARCHAR(15) to text
+	return obfuscateValue(u.ID, phone)[:15]
+}
+
+func obfuscateIdentityId(identity *Identity) string {
+	return obfuscateValue(identity.UserID, identity.Provider+":"+identity.ID)
 }
```

Comment:
nit: can consider making these functions as struct methods on the User or Identity instead

Reply:
I'd rather not busy the User model with the obfuscation stuff, it's only used here.